FirebaseSwiftUI/FirebaseAuthSwiftUI/Sources/Views/AuthPickerView.swift

```diff
@@ -15,46 +15,50 @@ public struct AuthPickerView {
 
 extension AuthPickerView: View {
   public var body: some View {
-    VStack {
-      Text(authService.string.authPickerTitle)
-        .font(.largeTitle)
-        .fontWeight(.bold)
-        .padding()
-      if authService.authenticationState == .authenticated {
-        SignedInView()
-      } else if authService.authView == .passwordRecovery {
-        PasswordRecoveryView()
-      } else if authService.authView == .emailLink {
-        EmailLinkView()
-      } else {
-        if authService.emailSignInEnabled {
-          Text(authService.authenticationFlow == .login ? authService.string
-            .emailLoginFlowLabel : authService.string.emailSignUpFlowLabel)
-          Divider()
-          EmailAuthView()
+    ScrollView {
+      VStack {
+        if authService.authView == .authPicker {
+          Text(authService.string.authPickerTitle)
+            .font(.largeTitle)
+            .fontWeight(.bold)
+            .padding()
         }
-        VStack {
-          authService.renderButtons()
-        }.padding(.horizontal)
-        if authService.emailSignInEnabled {
-          Divider()
-          HStack {
-            Text(authService
-              .authenticationFlow == .login ? authService.string.dontHaveAnAccountYetLabel :
-              authService.string.alreadyHaveAnAccountLabel)
-            Button(action: {
-              withAnimation {
-                switchFlow()
+        if authService.authenticationState == .authenticated {
```

Comment:
You could do this slightly more elegantly with a

```swift
switch authService.authView {
  case .passwordRecovery:
    // ...
  case .emailLink:
    // ...
}
```

Reply:
Using switch statement: [`c013f34` (#1259)](https://github.com/firebase/FirebaseUI-iOS/pull/1259/commits/c013f3403cd19816750586432174831cd3dc2142)

The only thing I don't like is including the default case at the end to satisfy compiler as we're not using `updatePassword` View in AuthPickerView.

Reply:
You can put `break` in the switch instead of `EmptyView`